### ISSUE TITLE
Feature: proportional font mode that understands markdown

### ISSUE THREAD

**Issue by itamarst** [jrblevin/markdown-mode#207]
It would be nice to able to use proportional fonts with markdown-mode. Merely enabling variable-pitch-mode breaks indentation, however. The solution is to have a monotype face for bullets, list items, indentation, and similar things - basically anything that indents consistently. Everything else (in practice all text) could be variable-pitch/proportional.

In order for this to work markdown-mode would need to expose a bunch more faces, e.g. a face for the indentation whitespace in list items.

**Comment by jrblevin** [comment]
Interesting idea—thanks. It's not something I'll probably get to soon, but I'll take a look at pull requests if anyone wants to give it a shot.

**Comment by hmelman** [comment]
I'm actually pretty happy using variable-pitch-mode with markdown-mode. But the new support of tables doesn't work well. Here's an idea of how to fix it based on a solution for org-mode. Could also be applied to code blocks. 

- https://stackoverflow.com/questions/3758139/variable-pitch-for-org-mode-fixed-pitch-for-tables
- https://yoo2080.wordpress.com/2013/05/30/monospace-font-in-tables-and-source-code-blocks-in-org-mode-proportional-font-in-other-parts/

**Comment by Fuco1** [comment]
I think having a font-lock face for the tables and the ability to customize them would go pretty far.  Is there any reason tables are not fontified?  I can make a PR that would apply some face to tables, that shouldn't be terribly difficult.

**Comment by jrblevin** [comment]
I agree that a distinct face for tables would be good. @Fuco1 If you are able to submit a PR that would be a big help.

**Comment by Fuco1** [comment]
All right, I've added an issue to my fork to track this: https://github.com/Fuco1/markdown-mode/issues/2 I'll try to get this done over the weekend.
